hudi-common/src/main/java/org/apache/hudi/common/table/read/FileGroupReaderSchemaHandler.java

```diff
@@ -162,7 +163,7 @@ protected InternalSchema doPruneInternalSchema(Schema requiredSchema, InternalSc
   }
 
   @VisibleForTesting
-  Schema generateRequiredSchema() {
+  public Schema generateRequiredSchema() {
```

Comment:
If public, should the `@VisibleForTesting` annotation be removed?

Reply:
This was actually just because the tests for this method were in the wrong class, I will update and revert this change to the visibility of the method